dcicutils/glacier_utils.py

```diff
@@ -0,0 +1,459 @@
+                2. s3_lifecycle_status to 'standard' by default
+                3. s3_lifecycle_category deleted (so lifecycle transition doesn't occur again)
+
+        :param atid: object resource path to update
+        :param status: status to replace at top level
+        :param s3_lifecycle_status: life cycle status to replace on file, typically standard as that is the default
+        :return: response from patch_metadata
+        """
+        return patch_metadata(
+            {
+                'status': status,
+                's3_lifecycle_status': s3_lifecycle_status
+            },
+            atid,
+            key=self.env_key, ff_env=self.env_name,
+            add_on='?delete_fields=s3_lifecycle_category'
+        )
+
+    def delete_glaciered_object_versions(self, bucket: str, key: str, delete_all_versions: bool = False) -> bool:
+        """ Deletes glaciered object versions of the given bucket/key, clearing all versions in glacier if the
+            delete_all_versions flag is specified
+
+        :param bucket: bucket location containing key
+        :param key: file name in s3 to delete
+        :param delete_all_versions: whether or not to delete all glacier versions or just the most recent one
+        :return: True if success or False if failed
+        """
+        try:
+            response = self.s3.list_object_versions(Bucket=bucket, Prefix=key)
+            versions = sorted(response.get('Versions', []), key=lambda x: x['LastModified'], reverse=True)
+            deleted = False
+            for v in versions:
+                if v.get('StorageClass') in GLACIER_CLASSES:
+                    response = self.s3.delete_object(Bucket=bucket, Key=key, VersionId=v.get('VersionId'))
+                    PRINT(f'Object {bucket}/{key} Glacier version {v.get("VersionId")} deleted:\n{response}')
+                    deleted = True
+                    if not delete_all_versions:
+                        break
+            if not deleted:
+                PRINT(f'No Glacier version found for object {bucket}/{key}')
+                return False
+            return True
+        except Exception as e:
+            PRINT(f'Error deleting Glacier versions of object {bucket}/{key}: {str(e)}')
+            return False
+
+    def copy_object_back_to_original_location(self, bucket: str, key: str, storage_class: str = 'STANDARD',
+                                              version_id: Union[str, None] = None) -> Union[dict, None]:
+        """ Reads the temporary location from the restored object and copies it back to the original location
+
+        :param bucket: bucket where object is stored
+        :param key: key within bucket where object is stored
+        :param storage_class: new storage class for this object
+        :param version_id: version of object, if applicable
+        :return: boolean whether the copy was successful
+        """
+        try:
```

Comment:
Consider adding a check here to ensure the object to copy is one that has been restored from glacier (if possible with metadata on the object), otherwise there's no reason to perform such a copy and accumulate costs.

Reply:
This isn't actually a copy, it's more like an overwrite (to change storage class), so no worries about increased cost (aside from those associated with the more expensive tier). It will fail if in an access tier that doesn't allow download I believe (ie: no restore active and in glacier or DA). This function is effectively just changing the storage tier.